### PR TITLE
Fix Native Rotation Android

### LINKED ISSUE #14161: Error when using rotation transform and "useNativeDriver: true" in Android
### Description

Animation do not work on Android when `useNativeDriver` is `true` and using a rotation transform setting the its angle as a `String` value. But it works as expected when setting it in radians using a `double` value. I usually set the rotation using a String as it is specified in the documentation, but in this case it crashes as it is expecting a double value.

The error I am getting is the following:

error screenshot

### Reproduction Steps and Sample Code

The following code sample would work either changing `useNativeDriver` to `false` or setting the rotation in radians.

```
  animateOpacity() {
    console.log('animation')
    Animated.timing(
      this.state.opacity_anim,
      {toValue: 1, duration: 600, useNativeDriver: true}
    ).start()
  }
  
  render() {
    return (
      <View style={styles.container}>
        {/* If the rotate transform is set in radians it works, but it doesn't when setting it in degrees with a string value. */}
        <Animated.View style={[
            styles.block,
            {
              transform: [{rotate: '90deg'}],
              opacity: this.state.opacity_anim
            }
        ]}>
          <Text style={styles.message}>Some text</Text>
        </Animated.View>
      </View>
    );
  }
```
I wrote an example in Expo as well:  https://snack.expo.io/SJGbutsl-

### Additional Information

* React Native version: v0.44.0, v0.43.0
* Platform: Android
* Development Operating System: macOS 10.12.5
* Dev tools: Android SDK 25.2.5, Android 7.0

### PR DESCRIPTION
**Pull request by scisci**
Fixes #14161
Android crashes in some cases if an animated transform config contains a string value, like a rotation.
This PR fixes that by ensuring all values sent to the native side are doubles. It adds `__transformDataType` to AnimatedTransform.js.

## Test Plan

Added integration test `ReactAndroid/src/androidText/js/AnimatedTransformTestModule.js` This test fails with the following error `INSTRUMENTATION_RESULT: longMsg=java.lang.ClassCastException: java.lang.String cannot be cast to java.lang.Double`, if the changes to AnimatedTransform.js are reverted.

## changelog

[Android] [Fixed] - Fixes Android crash on animated style with string rotation

### CODE REVIEW

**Comment by scisci** [comment]
@ janicduplessis

**Comment by henrikra** [comment]
I am waiting for this PR :)

File: Libraries/Animated/src/nodes/AnimatedTransform.js [hunk 1]
```diff
@@ -87,6 +87,22 @@ class AnimatedTransform extends AnimatedWithChildren {
     super.__detach();
   }
 
+  __transformDataType(value: any): number {
```
Comment (janicduplessis) [inline Libraries/Animated/src/nodes/AnimatedTransform.js]: Could you extract transformDataType to NativeAnimatedHelpers and use it here https://github.com/facebook/react-native/blob/4148976a83ab96029de1b26b515ff95d3cb58c10/Libraries/Animated/src/nodes/AnimatedInterpolation.js#L356 also.

Comment (scisci) [reply Libraries/Animated/src/nodes/AnimatedTransform.js]: OK I extracted to the common file. Interpolation works slightly differently (on an array) so it still needed the mapping function. This is from so long ago that I wasn't able to get the tests running again so maybe if you can check that it works.

**Comment by cpojer** [comment]
@scisci it seems there are some problems with your PR. Would you mind rebasing it on master and making sure the JS tests pass?

File: Libraries/Animated/src/nodes/AnimatedTransform.js (hunk 1 above)
Comment (janicduplessis) [reply Libraries/Animated/src/nodes/AnimatedTransform.js]: Yea sounds good, the main thing that is nice to extract out to be able to re-use is the deg to number conversion part. 👍

**Comment by scisci** [comment]
@cpojer, I rebased and tests are now passing

**Comment by janicduplessis** [comment]
@scisci Could you just address the lint issues that the bot pointed out? Especially the one about prettier since it might block landing this.

**Comment by scisci** [comment]
@janicduplessis I've fixed the eslint errors.

**Pull request closed without merging**